### ISSUE TITLE
[Context menu] component is inaccessible and has no warning in the docs #4170

### ISSUE THREAD

**Issue by grace-snow** [mui/base-ui#4171]
# Bug report

## Current behavior

There is no way to activate the Context Menu popup using a keyboard, screen reader, touch or speech input. There is no warning against using it or its accessibility limitations in the docs. 

This fails [WCAG 2.2 Success Criterion 2.1.1 Keyboard](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html)

## Expected behavior

Preferably, remove this component. It isn't necessary, and creates an inequitable experience between users of different modalities. The menu component already exists, which would be an accessible implementation of this functionality.

If it cannot be removed:
- either document that it is inaccessible to keyboard, screen reader, touch and speech input users and should not be used, including updating BaseUi's accessibility statement
- or update the component so that it has a named control to open the context menu 
- or update the component so that it is focusable and the usual keyboard shortcuts work (for example, <kbd>Control + Enter</kbd> on Mac or <kbd>Shift + F10</kbd> or dedicated menu keys on Windows)

## Reproducible example

Steps to reproduce: 
1. Navigate to the [Context menu](https://base-ui.com/react/components/context-menu) component demo
3. Attempt to open the menu using only a keyboard

## Base UI version

v1.2.0

## Which browser are you using?

Chrome 

## Which OS are you using?

Mac OS Tahoe 26.2

## Which assistive tech are you using (if applicable)?

Keyboard, VoiceOver, Apple Voice Control

## Additional context

The simplest solution would be to remove/retire this component from Base UI. 

Alternatively:
- Implement a focusable named region with appropriate keyboard shortcuts to open the context menu  
- Implement a separate visual indicator or additional control to open the context menu

**Comment by mj12albert** [comment]
Tested this a bit with `tabindex="0"` to the trigger:
- Windows Firefox & Edge: `Shift + F10` opens the context menu when the trigger is focused
- MacOS with "Full Keyboard Access" (OS setting) disabled
  - Firefox & Safari: `Ctrl + Enter` opens the menu
  - Chrome: keyboard shortcuts don't work
- MacOS with "Full Keyboard Access" enabled
  - Firefox: `Ctrl + Enter` opens the menu consistently, `Tab + M` works _sometimes_
  - Chrome: `Tab + M` opens the menu
  - Safari: neither work

I think we could add this recommendation to the docs:

> Implement a separate visual indicator or additional control to open the context menu

This would be like Slack's message context menu where the items are also available in the little toolbar in the corner:

<img width="320" alt="Image" src="https://github.com/user-attachments/assets/219f969e-269f-4253-b432-186b24a6c876" />

**Comment by grace-snow** [comment]
It would need a keyboard accessible control to _always_ be included with this component. It is not enough to have tabindex on a non-interactive element / region because that would still need an appropriate role and accessible name. 

With that in mind, could this component perhaps always include a control that offers a keyboard alternative? 

The documentation could then include guidance to move/reposition the control as needed, but note that it must always be included somewhere on the page/screen.

**Comment by grace-snow** [comment]
I have found this component implemented in a real world project now and it is causing **a severe accessibility issue** that makes some core functionality unavailable to keyboard and screen reader users. Is there a way to escalate this issue @mj12albert ?

**Comment by mj12albert** [comment]
@aarongarciah Do you think this is worth adding a demo in the docs for?

**Comment by aarongarciah** [comment]
> Preferably, remove this component. It isn't necessary, and creates an inequitable experience between users of different modalities. The menu component already exists, which would be an accessible implementation of this functionality.

> It would need a keyboard accessible control to _always_ be included with this component.

> I have found this component implemented in a real world project now and it is causing a **severe accessibility issue** that makes some core functionality unavailable to keyboard and screen reader users

@grace-snow as you probably know, using a component library that tries to be as accessible as possible doesn't remove the need for developers using such a library to exercise good judgment to make a website accessible.

In this case, a ContextMenu shouldn't be the only way of performing an action. A ContextMenu is an enhancement. It doesn't create inequitable experiences. To be honest, the suggestion of its removal makes no sense at all. I think it's better to keep issue discussions focused on realistic solutions, like updating the docs. 

I'll update the docs, adding some guidance.

**Comment by grace-snow** [comment]
Yes, that's fair. I apologise. I expect I wrote that from a point of frustration and should be more measured in recommendations. 

An update to the docs that emphasise caution/accessibility limitations with this component would be good. Especially if it recommends including a separate button to launch the same menu. 

Is there a slot in the component to make it easy to add a separate control already? 
Or could a variant be added to demonstrate how it can be done? That way there would be at least one accessible example available in the docs. 

CoPilot has implemented this component as-is in a recent project that left me unable to access some functionality. If the docs made the limitations clear and included an example showing how to include a separate control (which would allow speech input, keyboard and screen readers to open the context menu), then remediation will be simpler and this is less likely to be implemented inaccessibly in future.

**Comment by aarongarciah** [comment]
@grace-snow no need to apologize! We appreciate your involvement in the project and your desire to make things better.

I've opened this PR adding some usage guidelines and a demo on how to use ContextMenu alongside Menu https://github.com/mui/base-ui/pull/5121
